apps/website-25/src/components/courses/exercises/FreeTextResponse.tsx

```diff
@@ -42,23 +61,23 @@ const FreeTextResponse: React.FC<FreeTextResponseProps> = ({
       </div>
       <div className="free-text-response__options flex flex-col gap-2">
         <textarea
+          {...register('answer')}
           className={`free-text-response__textarea p-4${
-            isSaved ? ' free-text-response__textarea--saved container-active bg-[#63C96533] border-[#63C965] text-[#2A5D2A]' : ' container-lined'
+            (!isEditing && exerciseResponse) ? ' free-text-response__textarea--saved container-active bg-[#63C96533] border-[#63C965] text-[#2A5D2A]' : ' container-lined'
           }`}
           placeholder="Enter your answer here"
-          onChange={handleAnswerChange}
+          onChange={() => setIsEditing(true)}
         />
       </div>
       <CTALinkOrButton
         className="free-text-response__submit"
         variant="primary"
-        onClick={handleAnswerSubmit}
-        type="button"
+        type="submit"
       >
         Save
       </CTALinkOrButton>
-      {isSaved && <p className="free-text-response__saved-msg">Saved! 🎉</p>}
-    </div>
+      {(!isEditing && exerciseResponse) && <p className="free-text-response__saved-msg">Saved! 🎉</p>}
```

Comment:
consider: Similar to the above you can get `isSubmitSuccessful` and `isDirty` from `formState`, using the check `isSubmitSuccessful && !isDirty`

And in onSubmit you can call reset with keep values to mark things as 'clean'
```ts
reset({}, { keepValues: true });
```

This will save you from having to track isEditing yourself, and get you the behaviour I think you want: where by default nothing is shown. Only after submitting successfully, is the saved message shown. Once the user edits any of the fields it'll be marked as dirty and the message will hide.